azurerm/resource_arm_netapp_account_test.go

```diff
@@ -0,0 +1,376 @@
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMNetAppAccount_basicConfig(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-netapp-%d"
+  location = "%s"
+}
+
+resource "azurerm_netapp_account" "test" {
+  name                = "acctest-NetAppAccount-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+`, rInt, location, rInt)
+}
+
+func testAccAzureRMNetAppAccount_requiresImportConfig(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+resource "azurerm_netapp_account" "import" {
+  name                = "${azurerm_netapp_account.test.name}"
+  location            = "${azurerm_netapp_account.test.location}"
+  resource_group_name = "${azurerm_netapp_account.test.name}"
+}
+}
+`, testAccAzureRMNetAppAccount_basicConfig(rInt, location))
+}
+
+func testAccAzureRMNetAppAccount_completeConfig(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-netapp-%d"
+  location = "%s"
+}
+
+resource "azurerm_netapp_account" "test" {
+  name                = "acctest-NetAppAccount-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  active_directory {
+    username            = "aduser"
+    password            = "aduserpwd"
+    smb_server_name     = "SMBSERVER"
+    dns_servers         = ["1.2.3.4"]
+	domain              = "westcentralus.com"
+	organizational_unit = "OU=FirstLevel"
+  }
+}
+`, rInt, location, rInt)
+}
+
+func testAccAzureRMNetAppAccount_withIncorrectAccountNameConfig(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-netapp-%d"
+  location = "%s"
+}
+
+resource "azurerm_netapp_account" "test" {
+  name                = "@abc"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+}
+`, rInt, location)
+}
+
+func testAccAzureRMNetAppAccount_withIncorrectDomainNameConfig(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-netapp-%d"
+  location = "%s"
+}
+
+resource "azurerm_netapp_account" "test" {
+  name                = "acctest-NetAppAccount-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  active_directory {
+    username            = "aduser"
+    password            = "aduserpwd"
+    smb_server_name     = "SMBSERVER"
+    dns_servers         = ["1.2.3.4"]
+	domain              = "@abc"
+	organizational_unit = "OU=FirstLevel"
+  }
+}
+`, rInt, location, rInt)
+}
+
+func testAccAzureRMNetAppAccount_withIncorrectSMBServerNameConfig(rInt int, location string) string {
```

Comment:
We don't need a test config here, we can just test the validation function

Reply:
I've used validation.stringmatch so that i removed the test function here.